libopae/plugins/xfpga/sysfs.c

```diff
@@ -266,13 +266,22 @@ STATIC void find_resources(sysfs_fpga_region *region)
 	int num = -1;
 	char err[128] = {0};
 	regmatch_t matches[SYSFS_MAX_RESOURCES];
-	reg_res = regcomp(&re, SYSFS_FORMAT(sysfs_resource_fmt), REG_EXTENDED);
-	if (reg_res) {
-		regerror(reg_res, &re, err, 128);
-		FPGA_MSG("Error compiling regex: %s", err);
+
+	if (SYSFS_FORMAT(sysfs_resource_fmt)) {
+
+		reg_res = regcomp(&re, SYSFS_FORMAT(sysfs_resource_fmt), REG_EXTENDED);
+		if (reg_res) {
+			regerror(reg_res, &re, err, 128);
+			FPGA_MSG("Error compiling regex: %s", err);
```

Comment:
Shouldn't we return an error code here, similar to 401?

Reply:
fixed